web/src/main/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilter.java

```diff
@@ -284,6 +301,38 @@ private String getName(ServletContext context) {
         return name;
     }
 
+    /*This method is used to parse application name from the properties file*/
+    private String getApplicatioNameFromProperties() {
```

Comment:
We might want to add a unit test for this new behavior.

Reply:
@Dmitry-Matveev I agree that there should be test to test this behavior. However the WebRequestTrackingFilter does not implement any interface that we can mock using Mockito framework. The only way that we can write a testcase for this using JUnit can be actually reading from the file and doing the verification. Now that comes with a caveat. We need to store that file into our source code somewhere. If you have any other ideas on testing let me know.

Reply:
You can make `WebRequestTrackingFilter` implement an interface if it's required for tests since you control the code. Storing file within test project for test purposes is perfectly fine - you can consider such a file a 'test resource'.

Reply:
@Dmitry-Matveev interface will not help here, mocking interface makes only interface methods available for mocking. Better to make class not final and Mockito will be able to mock it.

Reply:
@gavlyukovskiy, makes sense. @dhaval24, could you please take a look at this approach?